lib/declare_schema/field_declaration_dsl.rb

```diff
@@ -28,13 +28,12 @@ def optimistic_lock
       field(:lock_version, :integer, default: 1, null: false)
     end
 
-    def field(name, type, *args)
-      options = args.extract_options!
-      @model.declare_field(name, type, *(args + [@options.merge(options)]))
+    def field(name, type, *args, **options)
+      @model.declare_field(name, type, *[*args, @options.merge(options)])
```

Comment:
Ah, `**` is so much nicer than `extract_options!`.

And I like this `[*` notation.

Reply:
This is all so much cleaner and easier to reason about!